File: src/main/java/org/aopbuddy/plugin/toolwindow/component/MyEditorTextField.java
package org.aopbuddy.plugin.toolwindow.component;

import com.intellij.openapi.application.ReadAction;
import com.intellij.openapi.application.WriteAction;
import com.intellij.openapi.command.WriteCommandAction;
import com.intellij.openapi.editor.Document;
import com.intellij.openapi.editor.EditorSettings;
import com.intellij.openapi.editor.ex.EditorEx;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.util.Computable;
import com.intellij.psi.PsiDocumentManager;
import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiFile;
import com.intellij.psi.PsiFileFactory;
import com.intellij.psi.codeStyle.CodeStyleManager;
import com.intellij.ui.EditorTextField;
import com.intellij.util.LocalTimeCounter;
import com.intellij.util.ui.JBUI;
import javax.swing.border.Border;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class MyEditorTextField extends EditorTextField {


  public MyEditorTextField(@NotNull Project project, @NotNull FileType fileType) {
    super(null, project, fileType, false, false);
  }


  public static void setupTextFieldEditor(@NotNull EditorEx editor) {
    EditorSettings settings = editor.getSettings();
    settings.setFoldingOutlineShown(true);
    settings.setLineNumbersShown(true);
    settings.setIndentGuidesShown(true);
    editor.setHorizontalScrollbarVisible(true);
    editor.setVerticalScrollbarVisible(true);
    editor.getScrollPane().setBorder(JBUI.Borders.empty());
  }

  protected EditorEx createEditor() {
    EditorEx editor = WriteAction.compute(() -> super.createEditor());
    initOneLineModePre(editor);
    setupTextFieldEditor(editor);
    return editor;
  }


  public void setBorder(Border border) {
    super.setBorder(JBUI.Borders.empty());
  }


  private void initOneLineModePre(@NotNull EditorEx editor) {
    editor.setOneLineMode(false);
    editor.setColorsScheme(editor.createBoundColorSchemeDelegate(null));
    editor.getSettings().setCaretRowShown(false);
  }

  public void setText(@Nullable String text, @NotNull FileType fileType) {
    Document document = createDocument(text, fileType);
    setNewDocumentAndFileType(fileType, document);
    PsiFile psiFile = ReadAction.compute(
        () -> PsiDocumentManager.getInstance(getProject()).getPsiFile(document));
    if (psiFile != null) {
      WriteCommandAction.runWriteCommandAction(getProject(),
          (Computable<PsiElement>) () -> CodeStyleManager.getInstance(getProject())
              .reformat(psiFile));
    }
  }

  public Document createDocument(@Nullable String text, @NotNull FileType fileType) {
    PsiFileFactory factory = PsiFileFactory.getInstance(getProject());
    long stamp = LocalTimeCounter.currentTime();
    PsiFile psiFile = factory.createFileFromText("aop-plugin", fileType,
        (text == null) ? "" : text, stamp, true, false);
    return PsiDocumentManager.getInstance(getProject()).getDocument(psiFile);
  }
}

File: src/main/kotlin/org/aopbuddy/plugin/toolwindow/component/MyEditorTextField.kt
package org.aopbuddy.plugin.toolwindow.component

import com.intellij.openapi.application.WriteAction
import com.intellij.openapi.editor.ex.EditorEx
import com.intellij.openapi.fileTypes.FileType
import com.intellij.openapi.project.Project
import com.intellij.ui.EditorTextField
import com.intellij.util.ui.JBUI
import org.jetbrains.annotations.NotNull

class MyEditorTextField(@NotNull project: Project, @NotNull fileType: FileType) : EditorTextField(null, project, fileType, false, false) {

    companion object {
        @JvmStatic
        fun setupEditor(@NotNull editor: EditorEx) {
            val settings = editor.settings
            settings.isFoldingOutlineShown = true
            settings.isLineNumbersShown = true
            settings.isIndentGuidesShown = true
            editor.scrollPane.setHorizontalScrollBarPolicy(javax.swing.ScrollPaneConstants.HORIZONTAL_SCROLLBAR_AS_NEEDED)
            editor.scrollPane.setVerticalScrollBarPolicy(javax.swing.ScrollPaneConstants.VERTICAL_SCROLLBAR_AS_NEEDED)
            editor.scrollPane.border = JBUI.Borders.empty()
        }
    }

    override fun createEditor(): EditorEx {
        val editor = WriteAction.compute<EditorEx, RuntimeException> { super.createEditor() }
        initOneLineModePre(editor)
        setupEditor(editor)
        return editor
    }

    override fun setBorder(border: javax.swing.border.Border?) {
        super.setBorder(JBUI.Borders.empty())
    }

    private fun initOneLineModePre(@NotNull editor: EditorEx) {
        editor.isOneLineMode = false
        editor.setColorsScheme(editor.createBoundColorSchemeDelegate(null))
        editor.settings.isCaretRowShown = false
    }
}